structuredheadings/dialogs/selectstyle.js

```diff
@@ -0,0 +1,35 @@
+CKEDITOR.dialog.add("selectStyle", function (editor) {
+
+  var styleOptions = [["Default", null]];
+
+  //Build out the style options
+  (function () {
+    for (var style in CKEDITOR.config.autonumberStyles) {
+      styleOptions.push([style, style]);
+    }
+  })();
+
+  return {
+    title: "Style Selection",
+    minWidth: 400,
+    minHeight: 200,
+    contents: [
+      {
+        id: "tab-style",
+        label: "Style Options",
+        elements: [
+          {
+            type: "radio",
+            id: "styleChooser",
+            items: styleOptions
+          }
+        ]
+      }
+    ],
+    onOk: function () {
+      var style = this.getValueOf("tab-style", "styleChooser");
+      editor.execCommand("setCurrentStyle", style);
+      editor.execCommand("reapplyStyle");
```

Comment:
I don't know if you want to check the return codes here (i don't think our commands can fail, yet).  But that might prevent cases where you reapply a style but the setCurrentStyle was not successful